src/server/graphql/mutations/joinIntegration.js

```diff
@@ -37,15 +39,25 @@ export default {
       throw new Error('You are already a part of this integration');
     }
 
-    // RESOLUTION
-    const updatedIntegration = await r.table(service).get(localId)
-      .update((doc) => ({
-        userIds: doc('userIds').append(userId).distinct()
-      }), {returnChanges: true})('changes')(0)('new_val').default(null);
+    const provider = await r.table('Provider')
+      .getAll(teamId, {index: 'teamIds'})
+      .filter({service, userId})
+      .nth(0)
+      .default(null);
+    if (!provider) {
+      throw new Error('You must first connect your account to the integration');
+    }
 
-    if (!updatedIntegration) {
-      throw new Error('Integration was already updated');
+    // RESOLUTION
+    // note: does not fail if they are already a member
+    if (service === GITHUB) {
```

Comment:
more and more i'm noticing that things like `addProvider` and `joinIntegration` have provider-specific login in them. for now there is so much shared logic that i think it's worth it to keep it the same, but i could be persuaded either way.